convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,255 @@
+        already_installed = re.search(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
+        if already_installed:
+            rhel_kernel_nevra = already_installed.group(1)
+            non_rhel_kernels = pkghandler.get_installed_pkgs_w_different_fingerprint(
+                system_info.fingerprints_rhel, "kernel"
+            )
+            for non_rhel_kernel in non_rhel_kernels:
+                # We're comparing to NEVRA since that's what yum/dnf prints out
+                if rhel_kernel_nevra == pkghandler.get_pkg_nevra(non_rhel_kernel):
+                    # If the installed kernel is from a third party (non-RHEL) and has the same NEVRA as the one available
+                    # from RHEL repos, it's necessary to install an older version RHEL kernel and the third party one will
+                    # be removed later in the conversion process. It's because yum/dnf is unable to reinstall a kernel.
+                    info_message = (
+                        "Conflict of kernels: One of the installed kernels"
+                        " has the same version as the latest RHEL kernel."
+                    )
+                    loggerinst.info("\n%s" % info_message)
+                    self.add_message(
+                        level="INFO",
+                        id="CONFLICT_OF_KERNELS",
+                        title="Conflict of installed kernel versions",
+                        description=info_message,
+                    )
+                    pkghandler.handle_no_newer_rhel_kernel_available()
+                    kernel_update_needed = True
+
+        if kernel_update_needed:
+            pkghandler.update_rhel_kernel()
+
+
+class VerifyRhelKernelInstalled(actions.Action):
+    id = "VERIFY_RHEL_KERNEL_INSTALLED"
+    dependencies = ("INSTALL_RHEL_KERNEL",)
+
+    def run(self):
+        """Verify that the RHEL kernel has been successfully installed and raise an ERROR if not"""
+        super(VerifyRhelKernelInstalled, self).run()
+
+        loggerinst.info("Verifying that RHEL kernel has been installed")
+        installed_rhel_kernels = pkghandler.get_installed_pkgs_by_fingerprint(
+            system_info.fingerprints_rhel, name="kernel"
+        )
+        if len(installed_rhel_kernels) <= 0:
+            self.set_result(
+                level="ERROR",
+                id="NO_RHEL_KERNEL_INSTALLED",
+                title="No RHEL kernel installed",
+                description="There is no RHEL kernel installed on the system.",
+                remediations="Verify that the repository used for installing kernel contains RHEL packages.",
+            )
+            return
+
+        loggerinst.info("RHEL kernel has been installed.")
+        self.add_message(
+            level="INFO",
+            id="RHEL_KERNEL_INSTALLED",
+            title="RHEL kernel installed",
+            description="The RHEL kernel has been installed successfully.",
+        )
+
```

Comment:
This block of message here looks a bit off, I think. We are not doing any kernel installations here, only checking it. 

Either we should update the message to reflect that the kernel is present on the system, or move it somewhere else...

Reply:
I'll update it to say RHEL kernel present on the system